x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/AbstractPhysicalOperationProviders.java

```diff
@@ -251,19 +257,22 @@ private void aggregatesToFactory(
                                 );
                             }
                         } else {
-                            sourceAttr = aggregateFunction.inputExpressions().stream().map(e -> {
-                                Attribute attr = Expressions.attribute(e);
+                            // extra dependencies like TS ones (that require a timestamp)
+                            for (Expression input : aggregateFunction.references()) {
+                                Attribute attr = Expressions.attribute(input);
```

Comment:
replace inputExpressions() with the existing references().
That method could be replaced in a separate PR or later on.

Reply:
I've removed this now - references does what's needed (looks at the dependencies and removes constants) tagging @dnhatn fyi.